pkg/logs/input/container/container.go

```diff
@@ -132,28 +138,36 @@ func (c *Container) isLabelMatch(labelFilter string) bool {
 // this feature is commonly named 'ad' or 'autodicovery'.
 const configPath = "com.datadoghq.ad.logs"
 
-// toSource converts a container to a source
-func (c *Container) toSource() *config.LogSource {
-	cfg := c.parseConfig()
+// toSource converts a container to a source,
+// returns an error if the parsing failed or processing rules are invalid.
+func (c *Container) toSource() (*config.LogSource, error) {
+	cfg, err := c.parseConfig()
+	if err != nil {
+		return nil, err
+	}
 	if cfg == nil {
-		return nil
+		return nil, nil
+	}
+	err = config.ValidateProcessingRules(cfg.ProcessingRules)
+	if err != nil {
+		return nil, err
 	}
-	return config.NewLogSource(configPath, cfg)
+	config.CompileProcessingRules(cfg.ProcessingRules)
+	return config.NewLogSource(configPath, cfg), nil
 }
 
 // parseConfig returns the config present in the container label 'com.datadoghq.ad.logs',
 // the config has to be conform with the format '[{...}]'.
-func (c *Container) parseConfig() *config.LogsConfig {
+func (c *Container) parseConfig() (*config.LogsConfig, error) {
 	label, exists := c.Labels[configPath]
 	if !exists {
-		return nil
+		return nil, nil
```

Comment:
Can we do better than this? eg. return `(*config.LogsConfig, bool)` akin to the semantics of checking whether an entry exists in a map. That would simplify the upstream logic. And you can move down the `log.Warnf` here, I don't think the error has other uses.